design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,474 @@
+
+| Secrets Provider variation                            | Business value                                               | Implementation status |
+| ----------------------------------------------------- | ------------------------------------------------------------ | --------------------- |
+| Run as **Init Container** and sync to **K8s Secrets** | Supply secrets on application load, based on K8s secrets     | Implemented           |
+| Run as **Job** and sync to **K8s Secrets**            | Supply secrets once for many applications, based on K8s secrets configurations | Milestone 1           |
+| Run as **Deployment** and sync to **K8s Secrets**     | Supply secrets for many applications and support rotation, based on K8s secrets | Future (Milestone 2)  |
+| Run as **Sidecar** and sync to **K8s Secrets**        | Support secrets rotation, based on K8s secrets               | Future (Milestone 2)  |
+| Run as **Init Container** and sync to **files**       | Supply secrets on application load, pushed as files to shared volume | Future (Milestone 3)  |
+| Run as **Sidecar** and sync to **files**              | Supply secrets for one application and support rotation, pushed as files to shared volume | Future (Milestone 3)  |
+
+These variations will be supported using the same Secrets Provider image, but the behavior will vary dynamically depending on the chosen deployment.
+
+*In this document we will focus on **Milestone 1**.*
+
+
+
+### Milestone 1: Serve K8s secrets to multiple applications once (no rotation)
+
+#### Design
+
+Run Secrets Provider as a **Job**.
+Once the Job's pod is spun up, it will authenticate to Conjur/DAP via authn-k8s.
+It will then fetch all the K8s secrets denoted with a specific label and update them with the Conjur secrets they require and upcome completion, terminate.
+
+###### How does a Job answer the requirements?
+
+Running as a **Job** allows separation from the applications' deployment and serve multiple applications at once.
+When Secrets Provider finishes updating the K8s secrets, the **Job** completes successfully and the pod terminates.
+
+###### What benefits does this solution provide?
+
+1. **Clarity**
+   Allowing customers to select the deployment to get the desired value
+2. **Maintainability**
+   Using one image for all variations simplifies code maintenance 
+3. **Reusability**
+   Using one image for all variations allows reuse of existing code across all variations
+4. **Extensibility**
+   Supporting various flows using the same solution opens the door for more secrets sources and targets
+
+###### What drawbacks does this solution have?
+
+**Job** is not an existing application identity granularity.
+
+As a result, one can define the host representing the Secrets Provider in Conjur policy using only **namespace** and/or **service account** granularities.
+
+###### How the drawbacks can be handled?
+
+To handle the missing **Job** granularity there are 2 options:
```

Comment:
@sigalsax - Why do we need a special granularity of job ? We can pass to Job input of namespace or of service account - and it should be good enough

Reply:
@InbalZilberman There are two options here either 1. creating a new application identity for job 2. or not. If not, use a dedicated service account with service account granularity. We wanted to pursue the latter because creating a new Job application identity will tack on time to delivery and might not be valuable to customer since they mainly use service account/namespace

Reply:
@InbalZilberman we currently support these k8s resources for application identity - https://docs.conjur.org/Latest/en/Content/Integrations/Kubernetes_AppIdentity.htm#SupportedKubernetesresources

As you can see, `Job` is not there. @sigalsax and @abrahamko are suggesting to not create such an application identity and let customers use the `service-account` application identity instead.